chromadb/api/types.py

```diff
@@ -194,6 +195,9 @@ def __call__(self: EmbeddingFunction[D], input: D) -> Embeddings:
 
         setattr(cls, "__call__", __call__)
 
+    def embed_with_retries(self, input: D, **retry_kwargs: Dict) -> Embeddings:
+        return retry(**retry_kwargs)(self.__call__)(input)
```

Comment:
Now that I think about this, this retry will work even if the errors from OPENAI are non transient correct? In cases where let say user max send limit for their Openai is reached no amount of retry is going to fix it until they update their spend limit. Do we want to retry for non transient errors? I guess consumers of chromaDb should be handling this no 🤔 

I am a user of chromaDB and what I have seen usually is Open AI waiting 600 secs and returning Timeouts. And literally so many folks complain about this error on their forum - https://community.openai.com/t/frequent-api-timeout-errors-recently/106903

We should also have a way to not wait 600 seconds and allow consumers to configure this. Ideally openAI should have given us a configuration option but there does not seem to be one. 

thoughts? @tazarov @beggers